components/properties/src/provider.rs

```diff
@@ -311,11 +314,128 @@ impl<'data, T: TrieValue> PropertyCodePointMapV1<'data, T> {
     }
 }
 
+/// `NormalizedPropertyNameBorrowed` is a fake type enabling us to use `#[zerovec::make_varule]`
+/// conveniently.
+/// This is a property name that can be "loose matched" as according to
+/// https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt
+///
+/// (matched case-insensitively in ASCII, ignoring underscores, whitespace, and hyphens)
+///
+/// This is expected to be ASCII, but we do not rely on this invariant anywhere except during
+/// datagen.
+#[derive(PartialEq, Eq)] // VarULE wants these to be byte equality
+#[zerovec::make_varule(NormalizedPropertyName)]
+#[cfg_attr(feature = "datagen", zerovec::derive(Serialize))]
+#[cfg_attr(feature = "serde", zerovec::derive(Deserialize))]
+#[zerovec::skip_derive(Ord)]
+pub struct NormalizedPropertyNameBorrowed<'a>(&'a [u8]);
+
+#[cfg(feature = "datagen")]
+impl<'a> serde::Serialize for NormalizedPropertyNameBorrowed<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::Error;
+        if serializer.is_human_readable() {
+            let s = str::from_utf8(self.0)
+                .map_err(|_| S::Error::custom("Attempted to datagen invalid string property"))?;
+            s.serialize(serializer)
+        } else {
+            self.0.serialize(serializer)
```

Comment:
Suggestion: Serde treats `&[u8]` as a "sequence of byte values" rather than "byte buffer"; use `serialize_bytes` instead. This is what we do in UnvalidatedStr, etc.